sys/mips/include/tls.h

```diff
@@ -38,32 +38,34 @@
  * TLS parameters
  */
 
-#ifdef __CHERI_PURE_CAPABILITY__
+#if defined(__CHERI_PURE_CAPABILITY__) || \
+    (defined(_KERNEL) && __has_feature(capabilities))
 #define TLS_TP_OFFSET	0
 #define TLS_DTP_OFFSET	0
 #else
 #define TLS_TP_OFFSET	0x7000
 #define TLS_DTP_OFFSET	0x8000
+#endif
+#ifdef COMPAT_FREEBSD32
```

Comment:
Does it actually make sense for these to be conditional on a kernel-only define when these aren't inside an `#ifdef _KERNEL`?

*he says as he spots the `COMPAT_CHERIABI` he probably wrote below...*

Reply:
I think it doesn't hurt.